packages/pentaho/src/Canvas/FloatingPanel/FloatingPanel.tsx

```diff
@@ -0,0 +1,225 @@
+import { forwardRef, useMemo } from "react";
+import { useResizeDetector } from "react-resize-detector";
+import {
+  ExtractNames,
+  HvActionGeneric,
+  HvActionsGeneric,
+  HvActionsGenericProps,
+  HvBaseProps,
+  HvPanel,
+  useControlled,
+  useDefaultProps,
+  useUniqueId,
+} from "@hitachivantara/uikit-react-core";
+
+import { HvCanvasTab } from "../Tab";
+import { HvCanvasTabs, HvCanvasTabsProps } from "../Tabs";
+import { staticClasses, useClasses } from "./FloatingPanel.styles";
+
+export { staticClasses as floatingPanelClasses };
+
+export type HvCanvasFloatingPanelClasses = ExtractNames<typeof useClasses>;
+
+export interface HvCanvasFloatingPanelProps extends HvBaseProps {
+  /** Open state of the floating panel. */
+  open?: boolean;
+  /** Minimize state of the floating panel tabs'. */
+  minimize?: boolean;
+  /** List of tabs visible on the panel. */
+  tabs: {
+    id: string | number;
+    title: React.ReactNode;
+  }[];
+  /** Id of the selected tab if it needs to be controlled. */
+  tab?: string | number;
+  /** Callback triggered when a tab changes/is clicked. */
+  onTabChange?: (
+    event: React.SyntheticEvent | null,
+    tabId: string | number | null,
+  ) => void;
+  /** Actions to be rendered in the left side of a tab. If more than one action is provided, a dropdown menu will be used. */
+  leftActions?: HvActionsGenericProps["actions"];
+  /** Actions to be rendered in the right side of a tab.If more than two actions are provided, a dropdown menu will be used to add the remaining actions. */
+  rightActions?: HvActionsGenericProps["actions"];
+  /** Callback triggered when a right or left action is clicked. */
+  onAction?: (
+    event: React.SyntheticEvent,
+    action: HvActionGeneric,
+    tabId: string | number,
+  ) => void;
+  /** A Jss Object used to override or extend the styles applied. */
+  classes?: HvCanvasFloatingPanelClasses;
+}
+
+/**
+ * A floating panel component to use in a canvas context.
+ */
+export const HvCanvasFloatingPanel = forwardRef<
+  HTMLDivElement,
+  HvCanvasFloatingPanelProps
+>((props, ref) => {
+  const {
+    id: idProp,
+    className,
+    children,
+    open,
+    tabs,
+    minimize,
+    leftActions,
+    rightActions,
+    tab: tabProp,
+    classes: classesProp,
+    onTabChange,
+    onAction,
+    ...others
+  } = useDefaultProps("HvCanvasFloatingPanel", props);
+
+  const { classes, cx } = useClasses(classesProp);
+
+  const id = useUniqueId(idProp);
+
+  // Tab resize detector: to position tab actions and set the panel top right border radius
```

Comment:
I'm using these `useResizeDetector` hooks to keep track of the tabs width, the left and right actions width, and the tab panel width in order to:
- Position the left and right actions on each tab as they have an absolute positioning for accessibility purposes: buttons can't be children of `button` or `tablist`;
- Know how much padding to give to the tab title since it depends on the size of the left and right actions provided by the user;
- Know the top right border radius of the tab panel since it can be 0 or 16 if the tabs occupy the full panel width or not, respectively;
- Detect if the tab content is overflowing to hide the buttons and only show the title: I took some liberties here as we don't have specs for these use case
Let me know if you think this is too much overhead and if you think that some of them should be dealt on the client side 🙏

Reply:
Looks good to me 👍 

I think the alternative would be to not use `button`, and have some kind of [complex nested navigation ](https://react-spectrum.adobe.com/react-aria/GridList.html) which is just too much work.